src/main/java/io/quarkus/search/app/hibernate/GuideLoadingBinder.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.search.app.hibernate;
+
+import java.net.URI;
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Named;
+
+import io.quarkus.search.app.entity.Guide;
+
+import org.hibernate.search.mapper.pojo.standalone.loading.LoadingTypeGroup;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassEntityLoader;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassEntitySink;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierLoader;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierSink;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassLoadingOptions;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassLoadingStrategy;
+import org.hibernate.search.mapper.pojo.standalone.loading.binding.EntityLoadingBinder;
+import org.hibernate.search.mapper.pojo.standalone.loading.binding.EntityLoadingBindingContext;
+
+@ApplicationScoped
+@Named("guideLoadingBinder")
+public class GuideLoadingBinder implements EntityLoadingBinder {
+
+    @Override
+    public void bind(EntityLoadingBindingContext entityLoadingBindingContext) {
+        entityLoadingBindingContext.massLoadingStrategy(Guide.class, new MassLoadingStrategy<Guide, URI>() {
+            @Override
+            public boolean equals(Object obj) {
+                return false;
```

Comment:
This seems... dangerous? Because `this.equals(this)` is now `false`...

Maybe just the identity-based equals/hash code from `super`?